lua/apisix.lua

```diff
@@ -143,6 +151,70 @@ function _M.http_ssl_phase()
 end
 
 
+local function parse_domain_in_up(up, ver)
+    local local_conf = core.config.local_conf()
+    local dns_resolver = local_conf and local_conf.apisix and
+                         local_conf.apisix.dns_resolver
+    local new_nodes = core.table.new(0, 8)
+
+    for addr, weight in pairs(up.value.nodes) do
+        local host, port = core.utils.parse_addr(addr)
+        if not ipmatcher.parse_ipv4(host) and
+           not ipmatcher.parse_ipv6(host) then
+            local ip_info = core.utils.dns_parse(dns_resolver, host)
```

Comment:
we should not drop the error msg of `dns_parse`. if the domain is invalid or timeout, what's the result? we should add test cases for that.